src/lib/reasoners/fun_sat.ml

```diff
@@ -1694,11 +1706,76 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
           "solved with backward!";
       raise e
 
+  let push env to_push =
+    if Options.get_tableaux_cdcl () then
+      Errors.run_error
+        (Errors.Unsupported_feature
+           "Incremental commands are not implemented in \
+            Tableaux(CDCL) solver ! \
+            Please use the Tableaux or CDLC SAT solvers instead"
+        );
+    let rec push_aux acc n =
+      if n <= 0 then acc
+      else
+        let b = E.fresh_name Ty.Tbool in
+        Stack.push b acc.incremental.stack_guard;
+
+        Stack.push (unit_facts acc) acc.unit_facts_cache;
+
+        let incremental =
```

Comment:
- Can probably be simplified only `incremental` field is updated and returned`
- We can probably use `Util.loop` for the recursion